atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/SweepStrategyManagers.java

```diff
@@ -84,9 +94,18 @@ public static SweepStrategyManager completelyConservative() {
         return tableRef -> SweepStrategy.from(TableMetadataPersistence.SweepStrategy.CONSERVATIVE);
     }
 
-    private static Map<TableReference, SweepStrategy> getSweepStrategies(KeyValueService kvs) {
-        return ImmutableMap.copyOf(
-                Maps.transformValues(kvs.getMetadataForTables(), SweepStrategyManagers::getSweepStrategy));
+    private static Map<TableReference, SweepStrategy> getSweepStrategiesForWarmingCache(
+            KeyValueService kvs, CacheWarming cacheWarming) {
+        switch (cacheWarming) {
+            case FULL:
+                return Maps.transformValues(kvs.getMetadataForTables(), SweepStrategyManagers::getSweepStrategy);
+            case LIMITED:
+                return Maps.transformValues(
+                        kvs.getLimitedMetadataForTables(50), SweepStrategyManagers::getSweepStrategy);
```

Comment:
How much value is there in having a separate `LIMITED` mode between `FULL` and `NONE`? If we set to `NONE` and just let the cache warm up on its own based on requests, what behavior would we see? How did we pick `50`? Are the first `50` tables actually the ones we want cached?

Reply:
50 was recommended I believe by @jeremyk-91 when this was discussed a while ago internally